test-framework/remote/src/main/java/org/keycloak/testframework/runonserver/RunOnServerSupplier.java

```diff
@@ -0,0 +1,62 @@
+package org.keycloak.testframework.runonserver;
+
+import org.apache.http.client.HttpClient;
+import org.keycloak.testframework.injection.InstanceContext;
+import org.keycloak.testframework.injection.LifeCycle;
+import org.keycloak.testframework.injection.RequestedInstance;
+import org.keycloak.testframework.injection.Supplier;
+import org.keycloak.testframework.injection.SupplierOrder;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.remote.RemoteProviders;
+import org.keycloak.testframework.server.KeycloakUrls;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RunOnServerSupplier implements Supplier<RunOnServerClient, InjectRunOnServer> {
+
+    @Override
+    public Class<InjectRunOnServer> getAnnotationClass() {
+        return InjectRunOnServer.class;
+    }
+
+    @Override
+    public Class<RunOnServerClient> getValueType() {
+        return RunOnServerClient.class;
+    }
+
+    @Override
+    public Set<Class<?>> dependencies() {
+        return Set.of(HttpClient.class, ManagedRealm.class, RemoteProviders.class, KeycloakUrls.class);
```

Comment:
```suggestion
        return Set.of(HttpClient.class, ManagedRealm.class, RemoteProviders.class);
```

Reply:
Updated - removed `dependencies` from Supplier as that might have been added by me by mistake (was something I experimented with, but never wired into the registry).